src/components/TimeBasedLineChart.js

```diff
@@ -0,0 +1,753 @@
+      tooltipPosition: false
+    });
+
+    this._removeSvgTooltipComponents();
+  }
+
+  _removeSvgTooltipComponents () {
+    const dots = d3.select(this.state.chartEl).selectAll('.dot-group');
+    const slices = d3.select(this.state.chartEl).selectAll('.slice-group');
+
+    slices.selectAll('.mx-time-based-line-chart-hover-rectangle').remove();
+    dots.selectAll('.mx-time-based-line-chart-hover-above').remove();
+    dots.selectAll('.mx-time-based-line-chart-hover-below').remove();
+    dots.selectAll('.mx-time-based-line-chart-hover-dot').remove();
+  }
+
+  _renderSvgTooltipComponents () {
+    const hoveredData = this.state.hoveredData;
+
+    const breakPointDate = moment.unix(this.props.breakPointDate);
+    const currentDate = moment.unix(hoveredData.timeStamp).startOf(this.props.rangeType);
+
+    const dash = currentDate.isAfter(breakPointDate) && this.props.dashedFutureLine ? '2, 2' : 'none';
+    const isBreakPointDate = currentDate.format('MM DD YYYY') === breakPointDate.format('MM DD YYYY');
+    const isDayRangeType = this.props.rangeType === 'day';
+    const dateFormat = isDayRangeType ? 'MM/DD' : 'MMM';
+
+    const dots = d3.select(this.state.chartEl).selectAll('.dot-group');
+    const slices = d3.select(this.state.chartEl).selectAll('.slice-group');
+
+    //Gray rectangle highlighting hovered section
+    slices.append('rect')
+      .attr('class', 'mx-time-based-line-chart-hover-rectangle')
+      .attr('x', () => {
+        return this._getXScaleValue(currentDate.unix()) - this._getSliceMiddle();
+      })
+      .attr('y', -10)
+      .attr('height', this.state.adjustedHeight)
+      .attr('width', () => {
+        return this._getSliceWidth(currentDate.unix());
+      })
+      .style('opacity', '0.025');
+
+    if (!this.props.staticXAxis) {
+      //Date Text
+      slices.append('text')
+        .attr('x', () => {
+          return this._getXScaleValue(currentDate.unix()) - this._getSliceMiddle() - 4;
+        })
+        .attr('y', this.state.adjustedHeight + 5)
+        .style('font-size', '10')
+        .style('opacity', '0.3')
+        .text(() => {
+          return isBreakPointDate ? null : currentDate.format(dateFormat);
+        });
+    }
+
+    //Credit balance line
+    if (hoveredData.above) {
+      dots.append('line')
+        .attr('class', 'mx-time-based-line-chart-hover-above')
+        .attr('x1', this._getXScaleValue(currentDate.unix()))
+        .attr('y1', this._getYScaleValue(hoveredData.value + hoveredData.above))
+        .attr('x2', this._getXScaleValue(currentDate.unix()))
+        .attr('y2', this._getYScaleValue(hoveredData.value))
+        .style('stroke-dasharray', dash)
+        .style('stroke-width', 3)
+        .style('stroke', '#30B53C')
+        .style('fill', 'none');
+    }
+
+    //Debit balance line
+    if (hoveredData.below) {
+      const below = hoveredData.value + hoveredData.below;
+      const belowValue = below < -4000 ? below + 500 : below;
```

Comment:
Used to prevent below line from crossing the bottom of the graph into the x axis tick labels.  I can see what I can do better here as well.